File: 2811-determine-the-minimum-sum-of-a-k-avoiding-array/determine-the-minimum-sum-of-a-k-avoiding-array.cpp
class Solution {
public:
    int minimumSum(int n, int k) {
        int sum=0,cnt=0,i=1;
        unordered_map<int,int>mp;
        while(cnt<n){
            if(mp.find(k-i)==mp.end()){
                cnt++;
                sum=sum+i;
                mp[i]++;
            }
            i++;
        }
        return sum;
    }
};

File: 2811-determine-the-minimum-sum-of-a-k-avoiding-array/determine-the-minimum-sum-of-a-k-avoiding-array.py
class Solution:
    def minimumSum(self, n: int, k: int) -> int:
        numSet = set()
        i = 1
        while len(numSet) != n:
            if k - i not in numSet:
                numSet.add(i)
            i += 1
        return sum(numSet)
        